contrib/utilities/indent

```diff
@@ -95,4 +95,35 @@ tab_to_space()
 export -f tab_to_space
 find tests include source unit_tests cookbooks benchmarks \( -name '*.cc' -o -name '*.h' -o -name '*.prm' -o -name '*.h.in' -o -name 'CMakeLists.txt' -o -name '*.cmake' \) -print | xargs -n 1 -P 10 -I {} bash -c 'tab_to_space "$@"' _ {}
 
+# Remove trailing whitespace from files
+remove_trailing_whitespace()
+{
+    f=$1
+    # awkward tab replacement because of OSX sed, do not change unless you test it on OSX
+    TAB=$'\t'
+    sed -e "s/[ $TAB]*$//"  $f >$f.tmp
+    diff -q $f $f.tmp >/dev/null || mv $f.tmp $f
+    rm -f $f.tmp
+}
+export -f remove_trailing_whitespace
+find tests include source unit_tests cookbooks benchmarks \( -name '*.cc' -o -name '*.h' -o -name '*.prm' -o -name '*.h.in' -o -name 'CMakeLists.txt' -o -name '*.cmake' \) -print | xargs -n 1 -P 10 -I {} bash -c 'remove_trailing_whitespace "$@"' _ {}
+
+# Ensure only a single newline at end of files
+ensure_single_trailing_newline()
+{
+  f=$1
+
+  # Remove newlines at end of file
+  # then add a newline if none exists
+  sed -e :a -e '/^\n*$/{$d;N;};/\n$/ba' $f >$f.tmpi
+  sed -e '$a\' $f.tmpi >$f.tmp
```

Comment:
Is this a mistake? I have no idea what these two lines actually do (maybe add a bit of commentary), but the second one with the `\'` that apparently does not escape the `'` seems particularly peculiar...

Reply:
It's not a mistake, but I agree it looks peculiar. I've now added several explanatory lines.

Reply:
What a bizarre statement. I wouldn't even know how this is interpreted by the shell. I guess using `'...'` instead of `"..."` means that the shell doesn't interpret things...